frontend/dockerfile/linter/generate.go

```diff
@@ -0,0 +1,137 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+type Rule struct {
+	Name        string
+	Description string
+}
+
+const tmplStr = `---
+title: {{.Rule.Name}}
+description: {{.Rule.Description}}
+---
+
+{{.Content}}
+`
+
+var destDir string
+
+func main() {
+	if len(os.Args) < 2 {
+		panic("Please provide a destination directory")
+	}
+	destDir = os.Args[1]
+	log.Printf("Destination directory: %s\n", destDir)
+	if err := run(destDir); err != nil {
+		panic(err)
+	}
+}
+
+func run(destDir string) error {
+	if err := os.MkdirAll(destDir, 0700); err != nil {
+		return err
+	}
+	rules, err := listRules()
+	if err != nil {
+		return err
+	}
+	tmpl, err := template.New("rule").Parse(tmplStr)
+	if err != nil {
+		return err
+	}
+	for _, rule := range rules {
+		if ok, err := genRuleDoc(rule, tmpl); err != nil {
+			return errors.Wrapf(err, "Error generating docs for %s", rule.Name)
+		} else if ok {
+			log.Printf("Docs generated for %s\n", rule.Name)
+		}
+	}
+	return nil
+}
+
+func genRuleDoc(rule Rule, tmpl *template.Template) (bool, error) {
+	mdfilename := fmt.Sprintf("docs/%s.md", rule.Name)
+	content, err := os.ReadFile(mdfilename)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			log.Printf("WARNING: Markdown file %q not found for %s\n", mdfilename, rule.Name)
+			return false, nil
+		}
```

Comment:
It needs a markdown file that contains examples for the rule. Atm we skip docs generation if the markdown file is not found but I think it should be required and fails if not found (cc @dvdksn)

Reply:
Updated to make validation fails if markdown files are not avilable.